newIDE/app/src/CommandPalette/CommandManager.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+type CmdHandler = () => void | Promise<void>;
+
+export type GlobalCommand = {|
+  displayText: string,
+  enabled: boolean,
+  handler: CmdHandler,
+|};
+
+export type UIGlobalCommand = {|
+  name: string,
+  ...GlobalCommand,
+|};
+
+export default class CommandManager {
+  globalCommands: { [string]: GlobalCommand };
+
+  constructor() {
+    this.globalCommands = {};
+    console.warn('Initialised command manager');
+  }
+
+  registerGlobal = (cmdName: string, cmdOpts: GlobalCommand) => {
```

Comment:
nitpicking again (sorry :D): `commandName` and `command`.

`cmdOpts` is unfortunately not a proper name because it's an abbreviation for "command options"... but this argument is not an "option for commands". It **is** a Command (with a big C, like the type).